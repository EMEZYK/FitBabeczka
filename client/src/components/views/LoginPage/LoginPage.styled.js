import styled from "styled-components";
import { theme } from "../../../theme/theme";
import { FlexWrapper } from "../../global-styles/Flex.styled";

export const SignInWrapper = styled(FlexWrapper)`
  background-color: #fffa;
  max-width: 30vw;
  height: 60vh;
  box-shadow: ${({ theme }) => theme.colors.card.loginBoxShadow};
`;

export const Title = styled.h2`
  font-size: ${({ fontSize }) =>
    fontSize ? fontSize : theme.fontsize.desktop.xxxl}px;
  margin: 0 0 2.5rem 0;
  /* text-transform: uppercase; */
  letter-spacing: 2px;
`;

export const LoginForm = styled.form`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  min-width: 24vw;
  min-height: 30vh;
  padding: 2.4rem;
  background-color: ${({ theme }) => theme.colors.common.body};
  border-radius: 0.3rem;
  box-shadow: ${({ theme }) => theme.colors.card.boxShadow};
`;

export const Input = styled.input`
  min-height: ${({ minHeight }) => (minHeight ? minHeight : "2.4rem")};
  text-align: left;
  padding: 0 0 0 0.6rem;
  margin: 0.6rem 0 0 0;
  border: ${({ border }) => (border ? border : "0.2px solid #aaa")};
  border-radius: 0.2rem;

  ::placeholder {
    text-transform: uppercase;
  }
`;

export const LoginButton = styled.button`
  min-height: 2.4rem;
  padding: 0.4rem;
  margin: 2rem 0 0 0;
  background-color: ${({ theme }) => theme.colors.button.secondary};
  color: #363434;
  font-size: ${({ fontSize }) =>
    fontSize ? fontSize : theme.fontsize.desktop.s}px;
  font-weight: ${({ theme }) => theme.weight.bold};
  text-transform: uppercase;
  letter-spacing: 1px;
  border: 0;
  border-radius: 0.2rem;
  box-shadow: ${({ theme }) => theme.colors.card.boxShadow};
  cursor: pointer;

  :hover {
    background-color: ${({ theme }) => theme.colors.button.active};
  }
`;
