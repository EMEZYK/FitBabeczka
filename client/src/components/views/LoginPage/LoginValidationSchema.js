import * as Yup from "yup";

const RegExp = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;

export const signInSchema = Yup.object({
  email: Yup.string().email("Invalid email").required("Email is required"),
  password: Yup.string()
    .required("Password is required")
    .min(8, "Password is too short - must contain  min 8 characters ")
    .matches(
      RegExp,
      "Password must contain 8 Characters, One Uppercase, One Lowercase, One Number and one special case Character"
    ),
});
