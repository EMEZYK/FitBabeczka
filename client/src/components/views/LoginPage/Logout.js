import React from "react";
import { toast, ToastContainer } from "react-toastify";
import { useCookies } from "react-cookie";
import { useNavigate } from "react-router-dom";
import "react-toastify/dist/ReactToastify.css";

export const Logout = () => {
  const [, removeCookie] = useCookies();

  const navigate = useNavigate();

  const logOut = async () => {
    try {
      await removeCookie("token");
      navigate("/user/login");
    } catch {
      return toast.error("Something went wrong, you can't logout");
    }
  };
  logOut();
  return (
    <>
      <ToastContainer />
    </>
  );
};
