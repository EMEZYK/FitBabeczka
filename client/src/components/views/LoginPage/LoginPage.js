import React from "react";
import {
  LoginForm,
  Title,
  LoginButton,
  SignInWrapper,
} from "./LoginPage.styled";
import { InputComponent } from "../../ui/Input/Input.styled";
import { FlexWrapper } from "../../global-styles/Flex.styled";
import { toast, ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { useNavigate } from "react-router-dom";
import { useCookies } from "react-cookie";
import { useFormik } from "formik";
import { LoginValidationSchema } from "./LoginValidationSchema";
import axios from "axios";
import jwt_decode from "jwt-decode";
import { BASE_URL } from "../../../context/recipes-context";

const LoginPage = ({ setUserId, setisLoggedIn, token, setTok }) => {
  const [, setToken] = useCookies(["token"]);
  let decodedToken;

  const signInFormHandler = useFormik({
    initialValues: {
      email: "",
      password: "",
    },
    validationSchema: LoginValidationSchema,
    onSubmit: (values) => JSON.stringify(values),
  });

  const navigate = useNavigate();

  const navigation = () => {
    navigate("/user/home");
  };

  const handleSignIn = signInFormHandler.handleSubmit;

  const onSubmitHandler = async (e) => {
    e.preventDefault();
    handleSignIn();

    await axios(
      {
        method: "POST",
        url: BASE_URL + "/admin/login",
        data: {
          email: signInFormHandler.values.email,
          password: signInFormHandler.values.password,
        },
      },
      { withCredentials: true }
    )
      .then((response) => {
        setToken("token", response.data.token);
        decodedToken = jwt_decode(response.data.token);

        const userId = decodedToken.userid;
        setUserId(userId);
        setisLoggedIn(true);
        navigation();
      })
      .catch(() => {
        return toast.error(
          "Nieudana próba logowania. Sprawdź wprowadzane dane"
        );
      });
  };

  return (
    <FlexWrapper width="100%">
      <SignInWrapper>
        <Title>Logowanie </Title>
        <LoginForm type="submit" onSubmit={onSubmitHandler}>
          <InputComponent
            type="email"
            id="email"
            data-testid="email"
            fieldName="email"
            placeholder="E-mail"
            value={signInFormHandler.values.email}
            onChange={signInFormHandler.handleChange}
            margin="0 0 0.6rem 0"
            required
          ></InputComponent>
          {signInFormHandler.errors.email ? (
            <div>{signInFormHandler.errors.email}</div>
          ) : null}
          <InputComponent
            type="password"
            id="password"
            data-testid="password"
            placeholder="Hasło"
            value={signInFormHandler.values.password}
            onChange={signInFormHandler.handleChange}
            required
          ></InputComponent>
          {signInFormHandler.errors.password ? (
            <div>{signInFormHandler.errors.password}</div>
          ) : null}
          <LoginButton type="submit">Zaloguj się</LoginButton>
          <ToastContainer />
        </LoginForm>
      </SignInWrapper>
    </FlexWrapper>
  );
};

export default LoginPage;
