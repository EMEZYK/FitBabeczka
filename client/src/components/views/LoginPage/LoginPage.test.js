import React from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom";
import LoginPage from "../LoginPage/LoginPage";

const LoginPageTest = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="*" element={<LoginPage policyTypes={false} />} />
      </Routes>
    </BrowserRouter>
  );
};

describe("LoginPage compponents", () => {
  afterEach(cleanup);

  test("Should take a snapshot',", () => {
    const { asFragment } = render(<LoginPageTest />);
    expect(asFragment(<LoginPageTest />)).toMatchSnapshot();
  });

  test("Verify login to the system", () => {
    render(<LoginPageTest />);
    expect(true).toBe(true);
  });

  test("Check if valid placeholder is displayed", () => {
    render(<LoginPageTest />);
    const emailInput = screen.queryByTestId("email");
    expect(emailInput).toHaveAttribute("placeholder", "E-mail");

    const passwordInput = screen.queryByTestId("password");
    expect(passwordInput).toHaveAttribute("placeholder", "Has≈Ço");
  });
});
