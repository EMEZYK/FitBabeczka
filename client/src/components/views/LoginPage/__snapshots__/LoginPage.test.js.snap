// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage compponents Should take a snapshot', 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ ciQwjP"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi hvLzQT egxkvU"
    >
      <h2
        class="sc-dkzDqf SUhDv"
      >
        Logowanie 
      </h2>
      <form
        class="sc-hKMtZM RlNcL"
        type="submit"
      >
        <input
          class="sc-gKXOVf kbCXOU"
          data-testid="email"
          id="email"
          placeholder="E-mail"
          required=""
          type="email"
          value=""
        />
        <input
          class="sc-gKXOVf cBzZEe"
          data-testid="password"
          id="password"
          placeholder="Hasło"
          required=""
          type="password"
          value=""
        />
        <button
          class="sc-jSMfEi hamXIW"
          type="submit"
        >
          Zaloguj się
        </button>
        <div
          class="Toastify"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
